### ISSUE TITLE
Process dictionary not available

### ISSUE THREAD

**Issue by amanmdesai** [HEP-FCC/fcc-tutorials#113]
Hi, 

I am facing issue while accessing: 
https://fcc-physics-events.web.cern.ch/fcc-physics-events/sharedFiles/FCCee_procDict_spring2021_IDEA.json

As given [here in the tutorial](https://hep-fcc.github.io/fcc-tutorials/fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.html#part-ii-analyse-with-fccanalyses:~:text=procDict%20%3D%20%22https%3A//fcc%2Dphysics%2Devents.web.cern.ch/fcc%2Dphysics%2Devents/sharedFiles/FCCee_procDict_spring2021_IDEA.json%22)

Could you please guide how to get the proc dict file. 

Thanks, Aman

**Comment by kjvbrt** [comment]
Hello @amanmdesai ,

can you try with the following address instead?
https://fcc-physics-events.web.cern.ch/fcc-physics-events/data/FCCDicts/FCCee_procDict_spring2021_IDEA.json

Best,
Juraj

**Comment by amanmdesai** [comment]
Hi @kjvbrt 
Thanks for the prompt reply. 

I am now able to access the file with the new link.

However, I unable to run  `fccanalysis final analysis_final.py`. 

A part of the error is as follows: 

```code
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)>
```

**Comment by kjvbrt** [comment]
Can you try to run the stages on `lxplus` or on RPM(redhat) based distribution?

**Comment by amanmdesai** [comment]
Unfortunately at the present moment I do not have access to lxplus or other machines.

**Comment by amanmdesai** [comment]
However, let me retry the stages with the updated link and will let you know

**Comment by kjvbrt** [comment]
I believe the problem likely lies in the differences between the key4hep stack and Ubuntu. FCCAnalyses uses `urllib` to get the json file from the server. Can you try to get the file in the Python REPL using one of the snippets?
https://docs.python.org/3/howto/urllib2.html
Try with and without sourcing of the FCCAnalyses environment.

**Comment by kjvbrt** [comment]
Anyway you should be able to just download the file and provide local path to the file and run the analysis stage.

**Comment by amanmdesai** [comment]
Hi @kjvbrt  thanks for your response. I am trying to re-do the tutorial for `fccanalysis` [as given here](https://hep-fcc.github.io/fcc-tutorials/fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.html) 

However, it seems at the moment `fccanalysis` command isnt working for me (earlier it worked for stage 1 and 2, but now the command is affected).   

I am able to use other utilities such as kkmce, etc. from the tutorial using the same setup and only `fccanalysis` command is affected. 

Error message when I call fccanalysis ... :

```error
Traceback (most recent call last):
  File "/cvmfs/sw.hsf.org/spackages6/fccanalyses/0.6.0/x86_64-centos7-gcc11.2.0-opt/qo45d/bin/fccanalysis", line 13, in <module>
    from config.Parsers import *
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 846, in exec_module
  File "<frozen importlib._bootstrap_external>", line 982, in get_code
  File "<frozen importlib._bootstrap_external>", line 1039, in get_data
OSError: [Errno 5] Input/output error: '/cvmfs/sw.hsf.org/spackages6/fccanalyses/0.6.0/x86_64-centos7-gcc11.2.0-opt/qo45d/python/config/__init__.py'
```

Could you please suggest me how to resolve this? 

Thanks, 
Aman

**Comment by amanmdesai** [comment]
Hi @kjvbrt 
I had issues with cvmfs setup, which are now resolved.


Going back to my original problem with procDict variable: As per your suggestions I tried urllib locally as well as with cvmfs. It opens successfully locally, but with cvmfs it fails. 

If I install the dict locally and try with that: I get the following error:
----> No procDict found: ===/afs/cern.ch/work/h/helsens/public/FCCDicts/FCCee_procDict_spring2021_IDEA.json===, exit

Basically it seems to attach "/afs/cern.ch/work/h/helsens/public/FCCDicts/" automatically in front of the .json file.

Any idea how to resolve this?

**Comment by amanmdesai** [comment]
Hi @kjvbrt 
The above message was due to issues with cvmfs installation which are now resolved. 

latest updates:
As per your suggestion I checked urllib. I could not use urllib within the cvmfs environment and hence tried to download the dict file locally and use it. 

I am able to run the file successfully. However this required additionally to update the `$FCCDICTSDIR` variable to my own system.  (without this the variable `$FCCDICTSDIR` directs to `/afs/cern.ch/work/h/helsens/public/FCCDicts/`). 

As an update to FCCAnalyses, I believe it would be nice if `$FCCDICTSDIR` is not set to a specific path. 
what do you think? 

Thanks very much for your help and time!

Thanks
Aman

**Comment by kjvbrt** [comment]
Hello @amanmdesai,

thanks for the thorough debuging. The key4hep stack is build to be used on top of CentOS 7, so running it on top of other OSes exposes lot of problems :) 

Of course, the `$FCCDICTSDIR` variable needs to be changed, maybe even the whole reliance on AFS could be removed.

Best,
Juraj

**Comment by amanmdesai** [comment]
> Of course, the $FCCDICTSDIR variable needs to be changed, maybe even the whole reliance on AFS could be removed.


Hi 

I just checked that the common defaults were updated in [FCCAnalyses](https://github.com/HEP-FCC/FCCAnalyses/blob/master/config/common_defaults.py) in October 2022 (v0.6.0 was released in Aug 2022). 

Hence the next software release might hopefully resolve this problem.
